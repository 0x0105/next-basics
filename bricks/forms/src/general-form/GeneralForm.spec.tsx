import React from "react";
import { shallow } from "enzyme";
import { GeneralForm, LegacyGeneralForm } from "./GeneralForm";

describe("GeneralForm", () => {
  it("should work", () => {
    const formElement = {} as any;
    const wrapper = shallow(
      <GeneralForm
        formElement={formElement}
        layout="inline"
        values={{
          username: "hello",
          date: "2019-10-01",
          time: "09:14:30",
          password: "world"
        }}
        valueTypes={{
          date: "moment",
          time: "moment|HH:mm:ss",
          password: ""
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("LegacyGeneralForm", () => {
  it("should assign `formUtils`", () => {
    const formElement = {} as any;
    const formUtils = {} as any;
    shallow(
      <LegacyGeneralForm
        formElement={formElement}
        form={formUtils}
        layout="inline"
        values={{
          username: "hello"
        }}
      />
    );
    expect(formElement.formUtils).toBe(formUtils);
  });
});
