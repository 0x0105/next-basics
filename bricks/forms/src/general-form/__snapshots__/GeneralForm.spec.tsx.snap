// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralForm should work 1`] = `
<LegacyGeneralForm
  form={
    Object {
      "getFieldDecorator": [Function],
      "getFieldError": [Function],
      "getFieldInstance": [Function],
      "getFieldProps": [Function],
      "getFieldValue": [Function],
      "getFieldsError": [Function],
      "getFieldsValue": [Function],
      "isFieldTouched": [Function],
      "isFieldValidating": [Function],
      "isFieldsTouched": [Function],
      "isFieldsValidating": [Function],
      "isSubmitting": [Function],
      "resetFields": [Function],
      "setFields": [Function],
      "setFieldsInitialValue": [Function],
      "setFieldsValue": [Function],
      "submit": [Function],
      "validateFields": [Function],
      "validateFieldsAndScroll": [Function],
    }
  }
  formElement={Object {}}
  layout="inline"
  valueTypes={
    Object {
      "date": "moment",
      "password": "",
      "time": "moment|HH:mm:ss",
    }
  }
  values={
    Object {
      "date": "2019-10-01",
      "password": "world",
      "time": "09:14:30",
      "username": "hello",
    }
  }
/>
`;
